### DISCUSSION TITLE
shape_index for obj/ply shapes

### DISCUSSION THREAD

**Discussion by GCChen97** [mitsuba-renderer/mitsuba3#314] [Feature requests]
Hi,

I try using AOV to output `shape_index` map and want to create masks for every shape according to the corresponding `shape_index`. But I couldn't find the relationship between the shapes and the shape indexes.

I found `shape_index` in shape `serialized` and wonder if this `shape_index` is the same one output by AOV. If so, adding a new property `shape_index` for obj/ply files would be convenient.

**Comment by Speierers** [comment]
This index corresponds to the shapes in the order they were instantiated when creating the scene. This doesn't have anything to do with the `shape_index` field in the `serialized` plugin.

**Reply by GCChen97** [reply]
Thanks for the comment. So it seems non-trivial to make AOV output the expected object label map.
